k8sutils/pkg/container/container.go

```diff
@@ -112,13 +124,23 @@ func AllContainersReady(pod *v1.Pod) bool {
 	if pod.Status.Phase != v1.PodRunning {
 		return false
 	}
+
+	skipStarted := isCronJobPod(pod)
+
 	// Iterate over all containers in the pod
 	// Return false if any container is:
 	// 1. Not Ready
 	// 2. Started is nil or false
 	for i := range pod.Status.ContainerStatuses {
 		containerStatus := &pod.Status.ContainerStatuses[i]
-		if !containerStatus.Ready || containerStatus.Started == nil || !*containerStatus.Started {
+
+		if !containerStatus.Ready {
+			return false
+		}
+
+		// For long-running pods (RestartPolicy=Always) ensure the container
+		// has actually entered the running state (`Started == true`).
+		if !skipStarted && containerStatus.Started != nil && !*containerStatus.Started {
```

Comment:
if `containerStatus.Started == nil`, it's considered as the container did not started, but this function will not `return false` (was like that before the change).

Is it intentional?